README.md

```diff
@@ -1,16 +1,695 @@
+      {"/": "bafkreibbz5pksvfjyima4x4mduqpmvql2l4gh5afaj4ktmw6rwompxynx4"}
+    ],
+    "run": {
+      "notify-bob": {
+        "with": "mailto://alice@example.com",
+        "do": "msg/send",
+        "inputs": [
+          {
+            "to": "bob@example.com",
+            "subject": "DNSLink for example.com",
+            "body": "Hello Bob!"
+          }
+        ]
+      },
+      "log-as-done": {
+        "with": "https://example.com/report"
+        "do": "crud/update"
+        "inputs": {
+          "from": "mailto://alice@exmaple.com",
+          "to": ["bob@exmaple.com"],
+          "event": "email-notification",
+          "value": {"ucan/promise": ["/", "notify-bob"]} // Pipelined promise
+        }
+      }
+    }
+  },
+  "sig": {"/": {"bytes:": "5vNn4--uTeGk_vayyPuNTYJ71Yr2nWkc6AkTv1QPWSgetpsu8SHegWoDakPVTdxkWb6nhVKAz6JdpgnjABppC7"}}
+}
+```
+
+# 4 Receipt
+
+An invocation receipt is an attestation of the output of an invocation. A receipt MUST be signed by the executor (the `aud` of the associated UCAN).
+
+**NB: a Receipt this does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+
+Receipts don't have their own version field. Receipts MUST use the same version as the invocation that they contain.
+
+## 4.1 Fields
+
+| Field  | Type            | Description                                                           | Required | Default |
+|--------|-----------------|-----------------------------------------------------------------------|----------|---------|
+| `inv`  | `&Invocation`   | CID of the Invocation that generated this response                    | Yes      |         |
+| `out`  | `{String: Any}` | The results of each call, the task's label. MAY contain sub-receipts. | Yes      |         |
+| `meta` | `Any`           | Non-normative extended fields                                         | No       | `null`  |
+
+### 4.1.1 Invocation
+
+The `inv` field MUST include a link to the Invocation that the Receipt is for.
+
+### 4.1.2 Output
+
+The `out` field MUST contain the output of steps of the call graph, indexed by the task name inside the invocation. If the invocation is the implicit `"*"`, then the base64 hash of the concatenation of the URI, Ability and extensional fields MUST be used.
+
+The `out` field MAY omit any tasks that have not yet completed, or results which are not public. In this case, it is RECOMMENDED that 
+
+### 4.1.3 Metadata Fields
+
+The metadata field MAY be omitted or used to contain additional data about the receipt. This field MAY be used for tags, commentary, trace information, and so on.
+
+## 4.2 IPLD Schema
+
+``` ipldsch
+type SignedReceipt struct {
+  rec Receipt (rename "ucan/receipt")
+  sig VarSig
+}
+
+type Receipt struct {
+  inv  &SignedInvocation
+  out  {String : Result}
+  meta Any (implicit Null)
+} 
+
+type Result union {
+  | Payload Any
+  | &SignedReceipt
```

Comment:
So it's either some value or another receipt ? Is receipt supposed to constitute an error ? I think some commentary here would be helpful